cli/src/diff_util.rs

```diff
@@ -152,10 +152,65 @@ fn diff_formats_from_args(
     Ok(formats)
 }
 
-fn default_diff_format(
+fn builtin_diff_format(name: &str, num_context_lines: Option<usize>) -> Option<DiffFormat> {
+    match name.as_ref() {
+        "summary" => Some(DiffFormat::Summary),
+        "types" => Some(DiffFormat::Types),
+        "git" => Some(DiffFormat::Git {
+            context: num_context_lines.unwrap_or(DEFAULT_CONTEXT_LINES),
+        }),
+        "color-words" => Some(DiffFormat::ColorWords {
+            context: num_context_lines.unwrap_or(DEFAULT_CONTEXT_LINES),
+        }),
+        "stat" => Some(DiffFormat::Stat),
+        _ => None,
+    }
+}
+
+fn diff_format_from_settings(
     settings: &UserSettings,
     num_context_lines: Option<usize>,
 ) -> Result<DiffFormat, config::ConfigError> {
+    let config = settings.config();
+    let format = if let Some(args) = config.get("ui.diff-formatter").optional()? {
+        if let CommandNameAndArgs::String(name) = &args {
+            // If setting is a single name starting with a colon it's treated as a built-in format. 
+            // It is still possible to use an external tool starting with a colon passing it as 
+            // an array rather than a string.
+            if name.starts_with(':') {
+              Some(builtin_diff_format(&name[1..], num_context_lines).ok_or_else(||
+                config::ConfigError::Message(format!("Unknown format setting for 'ui.diff-formatter', built-in formats are ':summary', ':types', ':git', ':color-words', and ':stat', or use an external tool")))?)
+                
+            } else {
+                let tool = merge_tools::get_external_tool_config(settings, name)?;
+                tool.map(|tool| DiffFormat::Tool(Box::new(tool)))
+            }
+        } else {
+            None
+        }
+        .or_else(|| {
+            Some(DiffFormat::Tool(Box::new(
+                ExternalMergeTool::with_diff_args(&args),
+            )))
+        })
+    } else {
+        Some(diff_format_from_deprecated_settings(
+            settings,
+            num_context_lines,
+        )?)
+    };
+
+    Ok(format.unwrap_or(DiffFormat::ColorWords {
+        context: num_context_lines.unwrap_or(DEFAULT_CONTEXT_LINES),
+    }))
+}
+
+fn diff_format_from_deprecated_settings(
```

Comment:
All the old behaviour is contained in this function, which can be removed if the time comes to go beyond deprecation.